### PR TITLE
feat: refactor built-in reviews to YAML config + pure markdown prompts

### PR DESCRIPTION
**Pull request by pacaplan**
## Summary

- Refactor built-in reviews from `built-in:` prefix convention to YAML config files with `builtin` attribute + pure markdown prompts
- Add `code-quality` as the first built-in review, shipped as `src/built-in-reviews/code-quality.{yml,md}`
- Update `init` command to generate YAML review config referencing the built-in
- Fix `detect` command to use the same change detection logic as `run` (rerun-mode awareness, auto-clean, fixBase resolution)
- Update docs and config references

## Commits

- `bc743a7` feat: add openspec proposal for built-in reviews support
- `a236c48` feat: add built-in review support with built-in:code-quality
- `b268daf` feat: refactor built-in reviews to use YAML config + pure markdown prompts
- `c8c7b17` chore: address PR review comments, switch project to builtin review
- `a745985` chore: update docs and config
- `133ca2c` fix: align detect command with run's change detection logic

## Test plan

- [x] `bun run build` passes
- [x] `bun test` passes
- [x] `bunx biome check src` passes
- [x] `agent-gauntlet run` passes (all gates)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Package-bundled "built-in" reviews supported via a YAML builtin reference; init now scaffolds .gauntlet/reviews/*.yml that reference built-ins (no local prompt.md created).

* **Documentation**
  * Config docs updated with a builtin field, examples, and mutual-exclusion rules among prompt sources.

* **Behavior Changes**
  * CLI preference merging and validation apply to built-in and file-based reviews; user review filenames using the reserved builtin prefix are rejected.

* **Tests**
  * Added comprehensive tests for loading, validation, merging, and edge cases.

### CODE REVIEW

File: .codescene/code-health-rules.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "matching_content_path": "test/**",
       "rules": [
         {
-          "name": "DRY Violations",
+          "name": "Code Duplication",
```
Comment (pacaplan) [inline .codescene/code-health-rules.json]: why did you change this?
